### PR TITLE
Make memory dump configurable.

### PR DESCRIPTION
**Pull request by hasenbanck**
This change makes the memory dump configurable by the user. It uses the same schema as the logging.

The default behavior is the current behavior to keep backward compability: Dumping main and bankable RAM.

I tried to minimize the usage of globals, but couldn't avoid them because of the current way the project is structured.

BTW: IMHO the ```SDL_PollEvent()``` handling should move out of ```video.c``` in the long run to better modulize the codebase in the future.

Implements #93 partialy.

### CODE REVIEW

**Comment by greg-king5** [comment]
Did you want to add a "memory.bin.new" file?

**Comment by hasenbanck** [comment]
@greg-king5 I don't understand your question. This PR allows you to configure the content of the memory dump.

The old behavior was to only save the main and banked RAM. This PR allows you to also dump the CPU registers, video RAM and video registers.

They are still generated either using the CTRL+S hotkey or reaching 0xFFFF in the PC. But this is the current behavior and this PR doesn't change that.

You can also still create multiple memory dumps using the CTRL hotkey multiple times. It will create "dump.bin", "dump-1.bin", "dump-2.bin" etc. file.

**Comment by greg-king5** [comment]
This PR modifies 7 files, _and adds one file_.  "memory.bin.new" is the file that's added.  But, it looks like it was something that was left over by the old code that you want to modify.  Perhaps, you committed it by mistake.

**Comment by hasenbanck** [comment]
Awwww .... mea culpa. That was a file to compare the output before and after the change. Fixed.

File: main.c [hunk 1]
```diff
@@ -101,6 +105,46 @@ label_for_address(uint16_t address)
...
+		return;
+	}
+
+    if (dump_cpu) {
+        fwrite(&a, sizeof(uint8_t), 1, f);
+        fwrite(&x, sizeof(uint8_t), 1, f);
+        fwrite(&y, sizeof(uint8_t), 1, f);
+        fwrite(&sp, sizeof(uint8_t), 1, f);
+        fwrite(&status, sizeof(uint8_t), 1, f);
+        fwrite(&pc, sizeof(uint16_t), 1, f);
```
Comment (mist64) [inline main.c]: this is not endianness safe. can you fputc() individual bytes?

File: main.c [hunk 2]
```diff
@@ -101,6 +105,46 @@ label_for_address(uint16_t address)
...
+		}
+		index++;
+	}
+	FILE *f = fopen(filename, "wb");
+	if (!f) {
+		printf("Cannot write to %s!\n", filename);
+		return;
+	}
+
+    if (dump_cpu) {
```
Comment (mist64) [inline main.c]: (spaces here, tabs everywhere else. the project uses tabs.)

**Comment by mist64** [comment]
Great, except for minor feedback. (And there are conflicts now.)

**Comment by hasenbanck** [comment]
@mist64 fixed the issues and rebased my branch.

**Comment by hasenbanck** [comment]
Wait ... Still has some megre conflicts? I just rebased ... oh Github ... let me look into it.

**Comment by hasenbanck** [comment]
Okay, all good now.

**Pull request merged**